static/app/views/eventsV2/table/queryField.tsx

```diff
@@ -176,12 +180,20 @@ class QueryField extends React.Component<Props> {
     this.triggerChange(newColumn);
   };
 
-  handleScalarParameterChange = (value: string) => {
-    const newColumn = cloneDeep(this.props.fieldValue);
-    if (newColumn.kind === 'function') {
-      newColumn.function[1] = value;
-    }
-    this.triggerChange(newColumn);
+  handleScalarParameterChange = (dataType: DataType) => {
```

Comment:
You can actually just chain the arrow functions like

```javascript
(dataType: DataType) => (value: string) => {
   // insert function body here
}
```

Reply:
ooh nice lol I was thinking in python decorator mode